java-frontend/src/test/files/se/ThisSuperNotNull.java

```diff
@@ -0,0 +1,23 @@
+class A {
+
+  void f() {
+    if (this == null) { // Noncompliant {{Change this condition so that it does not always evaluate to "false"}}
+
+    }
+    if (super == null) { // Noncompliant {{Change this condition so that it does not always evaluate to "false"}}
```

Comment:
Indeed, you are right this is not compilable. I replaced it with another usage of `super` identifier, even if that one is not explicitly demonstrating the change, since it is not directly possible to demonstrate an issue using `super` .